### PR TITLE
Resolve DAGMC Target Conflict and Improve Test Skips

### PR DESCRIPTION
**Pull request by ahnaf-tahmid-chowdhury**
## Summary

This PR refactors the PyNE build system to handle the conflict caused by the existing `dagmc` target while using Scikit Build Core. Additionally, I have improved the test suite by updating the pytest skips with clear reasons. Key changes include:

1. **Resolving DAGMC Target Conflict:**
   - In the legacy build system, we were managing the `dagmc` target using the line:
     ```cmake
     set_target_properties(_dagmc PROPERTIES OUTPUT_NAME dagmc)
     ```
     However, in the new approach, this method was not feasible. To resolve the conflict:
     - Renamed `dagmc.pyx` to `_dagmc.pyx`, which automatically creates a target `_dagmc` when building the Cython module.
     - Introduced a new `dagmc.py` that loads the `_dagmc` module.
  
2. **Test Suite Updates:**
   - Updated the test files to reflect the changes in the module name.
   - Improved the pytest skip statements by adding clear reasons for each skip, enhancing the understandability and maintainability of the test suite.

## Notes

This refactor ensures compatibility with the Scikit Build Core while preserving the integrity of the module structure and resolving conflicts with the `dagmc` target. The added reasons for pytest skips improve clarity during test runs.

### CODE REVIEW

File: tests/test_openmc_utils.py [hunk 1]
```diff
@@ -262,11 +262,11 @@ def test_get_result_error_from_openmc_sp():
 
 def test_create_meshtally():
     if not HAVE_PYMOAB or sys.version_info[0] == 2:
```
Comment (gonuke) [inline tests/test_openmc_utils.py]: I don't think we support python 2 anymore, so maybe we can get rid of that part of the test while we're here...

File: tests/test_dagmc.py [hunk 2]
```diff
@@ -5,20 +5,16 @@
...
-
 from pyne.mesh import HAVE_PYMOAB
-
 from pyne.utils import QAWarning
 
 warnings.simplefilter("ignore", QAWarning)
 
-try:
-    from pyne.mesh import Mesh
+from pyne.mesh import Mesh
```
Comment (gonuke) [inline tests/test_dagmc.py]: Could this be combine with the import above?

File: pyne/_dagmc.pyx [hunk 3]
```diff
@@ -9,7 +9,7 @@ from pyne.utils import QA_warn
 cimport numpy as np
 import numpy as np
 
-from pyne cimport cpp_dagmc_bridge
+cimport cpp_dagmc_bridge
```
Comment (gonuke) [inline pyne/_dagmc.pyx:12]: I'm not sure I understand why this change and/or how it works...

**Review by gonuke — CHANGES_REQUESTED** [review]
A few small questions - thanks for streamlining this for review @ahnaf-tahmid-chowdhury

File: pyne/_dagmc.pyx (hunk 3 above)
Comment (ahnaf-tahmid-chowdhury) [reply pyne/_dagmc.pyx:12]: scikit-build-core uses cmake to build cython modules, and I found that the build fails if we don't switch to this. It is unable to find cpp_dagmc_bridge.

**Comment by gonuke** [comment]
Note - there were many copies of the Python 2 test in that file....

**Comment by ahnaf-tahmid-chowdhury** [comment]
Sorry, just have removed all the python2 support with the recent commit 🙂

**Review by gonuke — APPROVED** [review]
Thanks for the py3 updates @ahnaf-tahmid-chowdhury
